File: back-end/app/controllers/imageController.js
import ImageService from "../services/imageService.js";

class ImageController {
  async getAllImages(req, res) {
    try {
      const images = await ImageService.getAllImages();
      res.json(images);
    } catch (error) {
      res
        .status(500)
        .json({ message: "Lỗi khi lấy danh sách hình ảnh", error });
    }
  }

  async getImageById(req, res) {
    try {
      const image = await ImageService.getImageById(req.params.id);
      if (!image)
        return res.status(404).json({ message: "Không tìm thấy hình ảnh" });
      res.json(image);
    } catch (error) {
      res.status(500).json({ message: "Lỗi khi lấy hình ảnh", error });
    }
  }

  async createImage(req, res) {
    try {
      const newImage = await ImageService.createImage(req.body);
      res.status(201).json(newImage);
    } catch (error) {
      res.status(500).json({ message: "Lỗi khi tạo hình ảnh", error });
    }
  }

  async updateImage(req, res) {
    try {
      const updatedImage = await ImageService.updateImage(
        req.params.id,
        req.body
      );
      if (!updatedImage)
        return res.status(404).json({ message: "Không tìm thấy hình ảnh" });
      res.json(updatedImage);
    } catch (error) {
      res.status(500).json({ message: "Lỗi khi cập nhật hình ảnh", error });
    }
  }

  async deleteImage(req, res) {
    try {
      const deletedImage = await ImageService.deleteImage(req.params.id);
      if (!deletedImage)
        return res.status(404).json({ message: "Không tìm thấy hình ảnh" });
      res.json({ message: "Xóa hình ảnh thành công" });
    } catch (error) {
      res.status(500).json({ message: "Lỗi khi xóa hình ảnh", error });
    }
  }
}

export default new ImageController();

File: server/src/main/java/com/instagram/server/controller/ImageController.java
package com.instagram.server.controller;

import com.instagram.server.entity.Image;
import com.instagram.server.service.ImageService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.List;
import java.util.Optional;

@RestController
@RequestMapping("/api/images")
public class ImageController {

    @Autowired
    private ImageService imageService;

    @GetMapping
    public List<Image> getAllImages() {
        return imageService.getAllImages();
    }

    @GetMapping("/{id}")
    public ResponseEntity<Image> getImageById(@PathVariable String id) {
        Optional<Image> image = imageService.getImageById(id);
        if (image.isPresent()) {
            return ResponseEntity.ok(image.get());
        } else {
            return ResponseEntity.notFound().build();
        }
    }

    @PostMapping
    public Image createImage(@RequestBody Image image) {
        return imageService.createImage(image);
    }

    @PutMapping("/{id}")
    public ResponseEntity<Image> updateImage(@PathVariable String id, @RequestBody Image imageDetails) {
        try {
            Image updatedImage = imageService.updateImage(id, imageDetails);
            return ResponseEntity.ok(updatedImage);
        } catch (RuntimeException e) {
            return ResponseEntity.notFound().build();
        }
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteImage(@PathVariable String id) {
        imageService.deleteImage(id);
        return ResponseEntity.noContent().build();
    }
}
